pandas/core/ops/__init__.py

```diff
@@ -605,14 +604,20 @@ def dispatch_to_extension_op(op, left, right):
     new_right = extract_array(right, extract_numpy=True)
 
     try:
-        res_values = op(new_left, new_right)
+        with warnings.catch_warnings():
```

Comment:
can you add a comment on the condition on what this is filtering. can we pre-empt the DeprecationWarning by a check?

Reply:
turns out this was the wrong/suboptimal place to do this check.  so this catch_warnings is now removed entirely and this is handled by checking for ABCExtensionArray before we get here